controllers/nodehealthcheck_controller.go

```diff
@@ -408,6 +418,13 @@ func (r *NodeHealthCheckReconciler) remediate(node *v1.Node, nhc *remediationv1a
 		return nil, nil
 	}
 
+	// Having a timeout also means we are using escalating remediations, for which we need to look at the "Progressing"
+	// condition, which can accelerate switching to the next remediator.
+	// So let's start watching the CRs, so we don't need to poll.
+	if err = r.addWatch(remediationCR); err != nil {
+		return nil, errors.Wrapf(err, "failed to add watch for %s", remediationCR.GroupVersionKind().String())
```

Comment:
IIUC watching for the condition status on the remediations isn't a "critical" service.
I'm not completely sure we want to fail the remediation in case because it fails.
Maybe log an event is enough ?
WDYT ?

Reply:
hm, unsure... on one hand you're right, it's not critical. But on the other hand I'm afraid no one will check events when things just work, just slower than it could be? I prefer to keep fail & retry, and be sure to get bugs reported... 🤔

Reply:
Fair enough, it's not something I feel strongly about and you do have a valid point.